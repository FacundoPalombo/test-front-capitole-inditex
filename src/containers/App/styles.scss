@use '../../assets/_normalize.scss';
@use '../../assets/_assets.scss';
