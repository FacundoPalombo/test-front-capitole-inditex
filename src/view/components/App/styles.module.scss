.heading {
  background-color: rebeccapurple;
  &__title {
    color: aliceblue;
  }
}
