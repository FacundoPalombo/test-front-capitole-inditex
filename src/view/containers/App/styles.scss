@import '../../../assets/_normalize.scss';
