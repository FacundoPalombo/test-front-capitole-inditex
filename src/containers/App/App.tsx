import React from 'react';
import './styles.scss';

import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
import { QueryClient } from '@tanstack/react-query';
import { PersistQueryClientProvider } from '@tanstack/react-query-persist-client';
import { createSyncStoragePersister } from '@tanstack/query-sync-storage-persister';

import { DAY_IN_MILLISECONDS } from '@App/utils/constants/various';

import Routes from '../routes/Routes';

//? Acá puse el stale time en un día tambien por que en el ejercicio "forzosamente" pide que se actualice la cache por un dia,
//? asi que "simule" una estrategia de cache-first configurando (stale-while-revalidata === max-age),
//? advierto esto NO es una buena practica en este caso de uso, solamente si se usa una estrategia de caching agresivo.
//? Usualmente, staleWhileRevalidate se realiza cada menos tiempo para poder refrescar la cache que del lado de la api que se vuelve "non-fresh"
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: DAY_IN_MILLISECONDS,
      cacheTime: DAY_IN_MILLISECONDS,
    },
  },
});

const persister = createSyncStoragePersister({
  storage: window.localStorage,
});

const App = () => {
  const routes = createBrowserRouter(Routes);
  return (
    <PersistQueryClientProvider
      client={queryClient}
      persistOptions={{ persister }}
    >
      <RouterProvider router={routes} />
      <ReactQueryDevtools initialIsOpen={false} />
    </PersistQueryClientProvider>
  );
};

export default App;
