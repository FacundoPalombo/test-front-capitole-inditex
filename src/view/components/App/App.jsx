import React from 'react';
import styles from './styles.module.scss';

const App = ({ example }) => {
  return (
    <main className={styles.heading}>
      <h1 className={styles.heading__title}>Hello {example}</h1>
    </main>
  );
};

export default App;
