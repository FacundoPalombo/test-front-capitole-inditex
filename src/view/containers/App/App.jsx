import React from 'react';
import './styles.scss';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import Routes from '../routes/Routes';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ReactQueryDevtools } from 'react-query/devtools';

const queryClient = new QueryClient();

const App = () => {
  const routes = createBrowserRouter(Routes);
  return (
    <QueryClientProvider client={queryClient}>
      <RouterProvider router={routes} />;
      <ReactQueryDevtools initialIsOpen={false} />
    </QueryClientProvider>
  );
};

export default App;
