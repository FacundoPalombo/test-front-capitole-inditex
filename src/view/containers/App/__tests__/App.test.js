/* eslint-disable react/display-name */

import React from 'react';
import App from '../App';
import { render } from '@testing-library/react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';

jest.mock('../../routes/Routes', () => () => <div>RoutesMockComponent</div>);
jest.mock('react-router-dom');

describe('App unit test', () => {
  afterAll(jest.clearAllMocks);
  it('should render ok with props', () => {
    const createBrowserRouterMock = jest.fn;
    const RouterMock = () => <div>RouterMock</div>;

    RouterProvider.mockImplementationOnce(RouterMock);
    createBrowserRouter.mockImplementationOnce(createBrowserRouterMock);

    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();

    expect(createBrowserRouter).toHaveBeenCalledTimes(1);
  });
});
