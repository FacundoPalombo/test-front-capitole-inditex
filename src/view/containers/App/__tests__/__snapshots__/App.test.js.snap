// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App integration test should render ok with props 1`] = `
<DocumentFragment>
  <main
    class="heading"
  >
    <h1
      class="heading__title"
    >
      Hello 22
    </h1>
  </main>
</DocumentFragment>
`;
